crates/sui-core/src/authority.rs

```diff
@@ -1005,6 +1005,46 @@ impl AuthorityState {
         DevInspectResults::new(effects, execution_result, self.module_cache.as_ref())
     }
 
+    pub async fn dev_inspect_move_call(
+        &self,
+        sender: SuiAddress,
+        move_call: MoveCall,
+        transaction_digest: TransactionDigest,
+    ) -> Result<DevInspectResults, anyhow::Error> {
+        let input_objects = move_call.input_objects();
+        let input_objects = transaction_input_checker::check_dev_inspect_input_objects(
+            &self.database,
+            input_objects,
+        )?;
+        let shared_object_refs = input_objects.filter_shared_objects();
+
+        let transaction_dependencies = input_objects.transaction_dependencies();
+        let temporary_store =
+            TemporaryStore::new(self.database.clone(), input_objects, transaction_digest);
+        let storage_gas_price = self
+            .database
+            .get_sui_system_state_object()?
+            .parameters
+            .storage_gas_price
+            .into();
+        let gas_status =
+            SuiGasStatus::new_with_budget(MAX_TX_GAS, storage_gas_price, storage_gas_price);
+        let (effects, execution_result) =
+            execution_engine::manual_execute_move_call::<execution_mode::DevInspect, _>(
+                shared_object_refs,
+                temporary_store,
+                sender,
+                move_call,
+                transaction_digest,
+                transaction_dependencies,
+                &self.move_vm,
+                gas_status,
+                self.epoch(),
+            );
+
+        DevInspectResults::new(effects, execution_result, self.module_cache.as_ref())
```

Comment:
wondering are `effects` required here, iiuc `effects` are about txns and move-call is only about `execution_result` ?
if we only need `execution_result`, is it possible to get rid of txn_digest?

Reply:
No, txn digest is used throughout the system. And the effects gives you information about objects created/modified/deleted.